### ISSUE TITLE
Transformer -  replicating results with no. of steps instead of epochs

### ISSUE THREAD

**Issue by srinivas-varadharajan** [tensorflow/models#5606]
### System information
- **What is the top-level directory of the model you are using**: models
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: RHEL 7.4
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**:  1.10.1
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A  training on CPU's
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

How is "batches per epoch" calculated in the Benchmarks Training Times mentioned in the document?
It is shown as 41365 for BIG transformer, but if you calculate using the code in the schedule.py, the results are different. 

Code segment that is used to calculate steps from epochs ,

total_num_tokens = NUM_EXAMPLES[mode] * self.max_length * num_epochs
return total_num_tokens // self.batch_size

if you choose the parameters from schedule.py and model_params.py,
NUM_EXAMPLES[train] = 4572160
max_length=256
batch_size=4096

Then the number of steps for 1 epoch would be 285760 and not  41365.
I understand the code segment I mentioned above is for TPU which uses static batch size, but I didn't find any code that did conversion for GPU/CPU. 
I'm trying to replicate results by mentioning the number of steps instead of number of epochs, any insights would be helpful.

Thanks

**Comment by billy19murahmi** [comment]
Hi,
@qlzh727 @robieta @k-w-w 
Did anyone get a chance to look at this?

**Comment by k-w-w** [comment]
The batch size for GPU/CPU works a bit differently than in TPU. As you noted, the batch size in TPU is static. Dynamic batching works by packing as many non-padding tokens into a single batch.

The batch_size for a dynamic batch is the maximum number of tokens in that batch. The examples in the dataset are grouped so that similar-length are in the same batch. Each group is then batched. For example, a single batch might consist of either:

* 256 examples with ~16 tokens in each example
* 128 examples with ~32 tokens in each example
* 64 examples with ~64 tokens in each example

Because of this dynamic batching, more examples can be packed together, instead of having each batch consist of 16 examples with 256 tokens in each example. Let me know if this makes sense or if you have any further questions!

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by amahendrakar** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
